### DISCUSSION TITLE
Add an example for listening to error and reconnection events

### DISCUSSION THREAD

**Discussion by tausifcreates** [aembke/fred.rs#177] [Ideas]
In the v6, there was example in the example/basics.rs file how to listen for error/reconnect events. 
```rs
let mut error_rx = client.on_error();
let mut reconnect_rx = client.on_reconnect();

tokio::spawn(async move {
    while let Ok(error) = error_rx.recv().await {
        println!("Redis client disconnected with error: {:?}", error);
    }
});

tokio::spawn(async move {
    while reconnect_rx.recv().await.is_ok() {
        println!("Successfully connected to Redis.");
    }
});
```

However, in the new release, this portion was removed and the example doesn't work (The api has changed)

Please consider adding an example again :)

**Comment by aembke** [comment]
Hi @tausifcreates, no problem, I added an example for this in the latest PR here: https://github.com/aembke/fred.rs/blob/main/examples/events.rs

Let me know if you have any other questions though.

**Comment by tausifcreates** [comment]
Thank you so much, @aembke! This is what I needed. It has been a pleasure to use Fred :)
